modular_skyrat/modules/xenos_skyrat_redo/code/xeno_types/sentinel.dm

```diff
@@ -0,0 +1,144 @@
+	desc = "Spits neurotoxin at someone, exhausting them."
+	icon_icon = 'modular_skyrat/modules/xenos_skyrat_redo/icons/xeno_actions.dmi'
+	button_icon_state = "neurospit_0"
+	plasma_cost = 25
+	/// A singular projectile? Use this one and leave acid_casing null
+	var/acid_projectile = /obj/projectile/neurotoxin/skyrat
+	/// You want it to be more like a shotgun style attack? Use this one and make acid_projectile null
+	var/acid_casing
+	/// Used in to_chat messages to the owner
+	var/projectile_name = "neurotoxin"
+	/// The base icon for the ability, so a red box can be put on it using _0 or _1
+	var/button_base_icon = "neurospit"
+	/// The sound that should be played when the xeno actually spits
+	var/spit_sound = 'modular_skyrat/modules/xenos_skyrat_redo/sound/alien_spitacid.ogg'
+	shared_cooldown = MOB_SHARED_COOLDOWN_3
+	cooldown_time = 5 SECONDS
+
+/datum/action/cooldown/alien/acid/skyrat/IsAvailable()
+	return ..() && isturf(owner.loc)
+
+/datum/action/cooldown/alien/acid/skyrat/set_click_ability(mob/on_who)
+	. = ..()
+	if(!.)
+		return
+
+	to_chat(on_who, span_notice("You prepare your [projectile_name] gland. <B>Left-click to fire at a target!</B>"))
+
+	button_icon_state = "[button_base_icon]_1"
+	UpdateButtons()
+	on_who.update_icons()
+
+/datum/action/cooldown/alien/acid/skyrat/unset_click_ability(mob/on_who, refund_cooldown = TRUE)
+	. = ..()
+	if(!.)
+		return
+
+	if(refund_cooldown)
+		to_chat(on_who, span_notice("You empty your [projectile_name] gland."))
+
+	button_icon_state = "[button_base_icon]_0"
+	UpdateButtons()
+	on_who.update_icons()
+
+/datum/action/cooldown/alien/acid/skyrat/InterceptClickOn(mob/living/caller, params, atom/target)
+	. = ..()
+	if(!.)
+		unset_click_ability(caller, refund_cooldown = FALSE)
+		return FALSE
+
+	var/turf/user_turf = caller.loc
+	var/turf/target_turf = get_step(caller, target.dir)
+	if(!isturf(target_turf))
+		return FALSE
+
+	var/modifiers = params2list(params)
+	caller.visible_message(
+		span_danger("[caller] spits [projectile_name]!"),
+		span_alertalien("You spit [projectile_name]."),
+	)
+
+	if(acid_projectile)
+		var/obj/projectile/spit_projectile = new acid_projectile(caller.loc)
+		spit_projectile.preparePixelProjectile(target, caller, modifiers)
+		spit_projectile.firer = caller
+		spit_projectile.fire()
+		playsound(caller, spit_sound, 100, TRUE, 5, 0.9)
+		caller.newtonian_move(get_dir(target_turf, user_turf))
+		return TRUE
+
+	if(acid_casing)
+		var/obj/item/ammo_casing/casing = new acid_casing(caller.loc)
+		playsound(caller, spit_sound, 100, TRUE, 5, 0.9)
+		casing.fire_casing(target, caller, null, null, null, ran_zone(), 0, caller)
+		caller.newtonian_move(get_dir(target_turf, user_turf))
+		return TRUE
+
+	CRASH("Neither acid_projectile or acid_casing are set on [caller]'s spit attack!")
+
+/datum/action/cooldown/alien/acid/skyrat/Activate(atom/target)
+	return TRUE
+
+/obj/projectile/neurotoxin/skyrat
+	name = "neurotoxin spit"
+	icon_state = "neurotoxin"
+	damage = 40
```

Comment:
the 40 stamina is a bit too strong, because it can be spammed, making it ironically almost as bad if not stronger then the soft-stun ones before (stamina crit is a 10s 'stun' that can be extended almost indefinitely with these)

Reply:
Keep in mind these projectiles use plasma very fast and are not going to be very effective against more than one person at a time.

Reply:
> Keep in mind these projectiles use plasma very fast and are not going to be very effective against more than one person at a time.

I tested them, it's not very expensive at all, you can alone handle up to 2-3 people with it, which is more then enough considering how strong the other xenos are